### PR TITLE
Investments: simplify constraints and remove InvestmentProperties trait

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Some simplifications regarding investment info/properties. [Slack conversation](https://kflabs.slack.com/archives/C04GJQAM9P0/p1695383388075129)

## Changes and Descriptions

- Removed the `InvestmentProperties` trait. Now, the type `InvestmentInfo` is used directly (which I think simplifies some parts)
- Removed the need for the `where` bound everywhere the investment pallets is used to determine the correct `InvestmentInfo` (this simplifies things for future integration tests)

### CODE REVIEW

File: pallets/investments/src/lib.rs [hunk 1]
```diff
@@ -595,7 +572,7 @@ where
 			amount,
 		})?;
 
-		let info = T::Accountant::info(investment_id).map_err(|_| Error::<T>::UnknownInvestment)?;
+		let _ = T::Accountant::info(investment_id).map_err(|_| Error::<T>::UnknownInvestment)?;
```
Comment (lemunozm) [inline pallets/investments/src/lib.rs:575]: Actually, this line could be removed and saved as read access to the database, but there is a test that checks for the error `UnknownInvestment`. If removed a future `Accountant::transfer()` fails with a different error.

File: pallets/investments/src/lib.rs [hunk 2]
```diff
@@ -662,7 +638,7 @@ where
 		who: T::AccountId,
 		investment_id: T::InvestmentId,
 	) -> DispatchResultWithPostInfo {
-		let info = T::Accountant::info(investment_id).map_err(|_| Error::<T>::UnknownInvestment)?;
+		let _ = T::Accountant::info(investment_id).map_err(|_| Error::<T>::UnknownInvestment)?;
```
Comment (lemunozm) [inline pallets/investments/src/lib.rs:641]: Same here

File: pallets/investments/src/lib.rs [hunk 3]
```diff
@@ -955,7 +928,7 @@ where
 			&mut total_order.amount,
 		)?;
 
-		T::Accountant::transfer(info.id(), send, recv, transfer_amount)
+		T::Accountant::transfer(investment_id, send, recv, transfer_amount)
```
Comment (lemunozm) [inline pallets/investments/src/lib.rs:931]: `info.id()` and `investment_id` always match with the same value

**Review by mustermeiszer — APPROVED** [review]
Changes look sane. But I am still thinking whether it is worth the refactor. Just am wondering whether we need to have that abstraction for `pallet-linked-pools`. 

But I we can go back and change it again if we want to.

**Comment by lemunozm** [comment]
Ok! Happy to revert some parts if needed in the future.

Note that the `InvestmentAccountant` trait is still generic over `InvestmentInfo`. So it is not fixed to always use the `InvestmentInfo` type.

**Pull request merged**